Library/Homebrew/test/cmd/list_spec.rb

```diff
@@ -4,19 +4,47 @@
 require "cmd/shared_examples/args_parse"
 
 RSpec.describe Homebrew::Cmd::List do
-  let(:formulae) { %w[bar foo qux] }
+  let(:formulae) { %w[bar qux foo] }
+  let(:casks) { %w[tex git sgb] }
 
   it_behaves_like "parseable arguments"
 
   it "prints all installed Formulae", :integration_test do
     formulae.each do |f|
       (HOMEBREW_CELLAR/f/"1.0/somedir").mkpath
     end
+    casks.each do |f|
+      (HOMEBREW_PREFIX/"Caskroom"/f/"42.0/somedir").mkpath
+    end
 
     expect { brew "list", "--formula" }
-      .to output("#{formulae.join("\n")}\n").to_stdout
+      .to output("#{formulae.sort.join("\n")}\n").to_stdout
       .and not_to_output.to_stderr
       .and be_a_success
+
+    expect { brew "list", "--cask" }
+      .to output("#{casks.sort.join("\n")}\n").to_stdout
+      .and not_to_output.to_stderr
+      .and be_a_success
+
+    expect { brew "list", "--formula", "--version" }
+      .to output("#{formulae.sort.map { |name| "#{name} 1.0" }.join("\n")}\n").to_stdout
+      .and not_to_output.to_stderr
+      .and be_a_success
+
+    # expect { brew "list", "--cask", "--version" }
+    #   .to output("#{casks.sort.map { |name| "#{name} 42.0" }.join("\n")}\n").to_stdout
+    #   .and not_to_output.to_stderr
+    #   .and be_a_success
+
+    # expect { brew "list", "--version" }
+    #   .to output(<<~OUTPUT,
+    #     #{formulae.sort.map { |name| "#{name} 1.0" }.join("\n")}
+    #     #{casks.sort.map { |name| "#{name} 42.0" }.join("\n")}
+    #   OUTPUT
+    #             ).to_stdout
+    #   .and not_to_output.to_stderr
+    #   .and be_a_success
```

Comment:
Let's not keep commented test cases here. Either fix them so they can be used or remove them. Thanks!

Reply:
I'm still wondering why the `brew lists --cask` testcase seems to be happy with my `:casks` array (and paths), but `brew list --cask --version` trips over the empty list. Any ideas?

Reply:
Correct me if I'm wrong, but after poking around in the codebase, it appears that `brew list --cask` “succeeds”, because `Cask::Caskroom.casks` [catches the exception](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cask/caskroom.rb#L62), while `Cask::List.list_casks` [bombs](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cask/list.rb#L12).

Reply:
This might also be the reason why `brew list` executes an order of magnitude faster than `brew list --version`.